cosmos/operators/virtualenv.py

```diff
@@ -102,11 +117,110 @@ def run_subprocess(self, command: list[str], env: dict[str, str], cwd: str) -> F
         )
         return subprocess_result
 
+    def clean_dir_if_temporary(self) -> None:
+        """
+        Delete the virtualenv directory if it is temporary.
+        """
+        if self.is_virtualenv_dir_temporary and self.virtualenv_dir and self.virtualenv_dir.exists():
+            self.log.info(f"Deleting the Python virtualenv {self.virtualenv_dir}")
+            shutil.rmtree(str(self.virtualenv_dir), ignore_errors=True)
+
     def execute(self, context: Context) -> None:
-        output = super().execute(context)
-        if self._venv_tmp_dir:
-            self._venv_tmp_dir.cleanup()
-        logger.info(output)
+        try:
+            super().execute(context)
+        finally:
+            self.clean_dir_if_temporary()
+
+    def on_kill(self) -> None:
+        self.clean_dir_if_temporary()
+
+    def prepare_virtualenv(self) -> str:
```

Comment:
can we have some identifier here to differentiate with the `prepare_virtualenv` method from `airflow.utils.python_virtual_env` below in `py_bin = prepare_virtualenv(....`? 

Maybe we could have a leading underscore for this method? or something else to avoid potential confusion while reading the code. My bad, but I first thought we are recursively calling the same method 🙃

Reply:
Addressed in b465b2a21e7ee21a6017e379a16a607ea2259859